src/core.ts

```diff
@@ -97,29 +98,65 @@ export interface Options {
   killSignal?:    NodeJS.Signals
   halt?:          boolean
 }
-// prettier-ignore
-export const defaults: Options = {
-  [CWD]:          process.cwd(),
-  [SYNC]:         false,
-  verbose:        false,
-  env:            process.env,
-  sync:           false,
-  shell:          true,
-  stdio:          'pipe',
-  nothrow:        false,
-  quiet:          false,
-  prefix:         '',
-  postfix:        '',
-  detached:       false,
-  preferLocal:    false,
-  spawn,
-  spawnSync,
-  log,
-  kill,
-  killSignal:     SIGTERM,
-  timeoutSignal:  SIGTERM,
+
+export function getZxDefaults(
+  defs: Options,
+  prefix: string = 'ZX_',
+  env = process.env
+) {
+  const types: Record<PropertyKey, Array<'string' | 'boolean'>> = {
+    cwd: ['string'],
+    shell: ['string', 'boolean'],
+    halt: ['boolean'],
+    preferLocal: ['string', 'boolean'],
+    input: ['string'],
```

Comment:
I guess we should restrict the accepted env opts: `sync, nothrow, input, halt, stdio` can break the script logic, while `verbose, shell, quiet, preferLocal` just change its behaviour. Moreover, `timeout` does not belong to `defaults`, but it may be reasonable to set externally. 

```ts
const allowed = new Set(['timeout', 'shell', ...])
// ...
if (allowed.has(_k)) {
}
```
And if somebody is setting `ZX_CWD=true`, wrong typecast to bool is the least of problems. So maybe we can skip this check.

Reply:
Plz clarify what you suggest:

* determine only one list of acceptable properties (`verbose, shell, quiet, preferLocal`) and try it extract from env with type-checking

* determine two lists:

  * acceptable - extract w/o type check
  * unacceptable - extract with strict type check

Reply:
How about this?

```ts
(defs: Options, prefix = 'ZX_', env = process.env) => Object.entries(env).reduce<Options>((m, [k, v]) => {
  const allowed = new Set(['preferLocal', 'detached', 'verbose', 'quiet', 'timeout', 'timeoutSignal', 'prefix', 'postfix'])
  if (v && k.startsWith(prefix)) {
    const _k = snakeToCamel(key.slice(prefix.length))
    const _v = ({true: true, false: false})[v.toLowerCase()] ?? v
    if (allowed.has(_k))
      m[_k] =  _v
  }
  return m
}, defs)
```

Reply:
You insistently suggest pass Options to `reduce` method - it isn't work here `m[_k] =  _v`. Since `_k` is random key of Options, `m[_k]` has type `never` as union of `string`, `boolean`, `StdioInput` and other values of Options.
![image](https://github.com/user-attachments/assets/f0607010-b7f6-47bc-9b58-a19231472319)

Reply:
> How about this?
> 
> ```ts
> (defs: Options, prefix = 'ZX_', env = process.env) => Object.entries(env).reduce<Options>((m, [k, v]) => {
>   const allowed = new Set(['preferLocal', 'detached', 'verbose', 'quiet', 'timeout', 'timeoutSignal', 'prefix', 'postfix'])
>   if (v && k.startsWith(prefix)) {
>     const _k = snakeToCamel(key.slice(prefix.length))
>     const _v = ({true: true, false: false})[v.toLowerCase()] ?? v
>     if (allowed.has(_k))
>       m[_k] =  _v
>   }
>   return m
> }, defs)
> ```

In that case we can assign to `verbose` and `quiet` some string value. It's not good, in my opinion.
So, I combined 'allowed' and type-check together into `types` and use only it.